tests/FunctionCallTest.php

```diff
@@ -1300,18 +1315,46 @@ function takesMatches(array $matches) : void {}
 
                     takesMatches($matches);',
             ],
+            'pregMatchWithOffset2' => [
+                '<?php
+                    $r = preg_match("{foo}", "foo", $matches, 0, 10);',
+                'assertions' => [
+                    '$r===' => '0|1|false',
+                    '$matches===' => 'array<array-key, string>',
+                ],
+            ],
             'pregMatchWithFlags' => [
                 '<?php
                     function takesInt(int $i) : void {}
 
                     if (preg_match("{foo}", "this is foo", $matches, PREG_OFFSET_CAPTURE)) {
-                        /**
-                         * @psalm-suppress MixedArrayAccess
-                         * @psalm-suppress MixedArgument
-                         */
                         takesInt($matches[0][1]);
                     }',
             ],
+            'pregMatchWithFlagOffsetCapture' => [
+                '<?php
+                    $r = preg_match("{foo}", "foo", $matches, PREG_OFFSET_CAPTURE);',
+                'assertions' => [
+                    '$r===' => '0|1|false',
+                    '$matches===' => 'array<array-key, array{string, int}>',
+                ],
+            ],
+            'PHP72-pregMatchWithFlagUnmatchedAsNull' => [
```

Comment:
PHP72 here means it would be skipped when Psalm itself is running on 7.1. To specify analysis target version you need to put it into an element of the test case array, e.g.:
https://github.com/vimeo/psalm/blob/ad23ab3405d11ab95b06602afd8d4fb833ad657a/tests/ClosureTest.php#L53-L64

Reply:
Hehe, that happens, when there's no documentation :stuck_out_tongue_winking_eye:

Hmm, thinking about it, do I need to specify a version at all in this case?!
I added it as ``PREG_UNMATCHED_AS_NULL`` was added in 7.2

Reply:
@weirdan Hmm, after thinking about it, I believe it is ok as it is then. Tell me if I have an error in reasoning.

It’s a test and the test is executed by Psalm. The test would fail running on PHP 7.1 as ``PREG_UNMATCHED_AS_NULL`` is not defined. So it has to be skipped?!

Reply:
> The test would fail running on PHP 7.1 as PREG_UNMATCHED_AS_NULL is not defined...
... if Psalm uses reflection to figure out constant types. I'm not 100% it does (it looks quite possible though).

Reply:
The oldest PHP version I have to to test atm is 7.4…
At least ``PREG_UNMATCHED_AS_NULL`` can only be found once in the source and that´s no definition. So I’m almost sure Psalm uses reflection to get it.

Reply:
Indeed, with PHPUnit 7 and PHP 7.1 I get the following:

```console
$ vendor/bin/phpunit tests/FunctionCallTest.php --filter unmatchedasnull
PHPUnit 7.5.20 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.1.33 with Xdebug 2.9.8
Configuration: /home/weirdan/src/psalm/psalm/phpunit.xml.dist
Random seed:   1638306885

  Warning - The configuration file did not pass validation!
  The following problems have been detected:

  Line 14:
  - Element 'coverage': This element is not expected.

  Test results may not be as expected.


EE                                                                  2 / 2 (100%)

Time: 1.08 seconds, Memory: 36.00 MB

There were 2 errors:

1) Psalm\Tests\FunctionCallTest::testValidCode with data set "pregMatchWithFlagOffsetCaptureAndUnmatchedAsNull" ('<?php\n                    $r...NULL);', array('0|1|false', 'array<array-key, array{null|s... int}>'))
Psalm\Exception\CodeException: UndefinedConstant - src/somefile.php:2:85 - Const PREG_UNMATCHED_AS_NULL is not defined

/home/weirdan/src/psalm/psalm/src/Psalm/IssueBuffer.php:270
/home/weirdan/src/psalm/psalm/src/Psalm/IssueBuffer.php:114
/home/weirdan/src/psalm/psalm/src/Psalm/Internal/Analyzer/Statements/Expression/Fetch/ConstFetchAnalyzer.php:93
/home/weirdan/src/psalm/psalm/src/Psalm/Internal/Analyzer/Statements/ExpressionAnalyzer.php:162
/home/weirdan/src/psalm/psalm/src/Psalm/Internal/Analyzer/Statements/ExpressionAnalyzer.php:51
/home/weirdan/src/psalm/psalm/src/Psalm/Internal/Analyzer/Statements/Expression/BinaryOpAnalyzer.php:111
/home/weirdan/src/psalm/psalm/src/Psalm/Internal/Analyzer/Statements/ExpressionAnalyzer.php:238
/home/weirdan/src/psalm/psalm/src/Psalm/Internal/Analyzer/Statements/ExpressionAnalyzer.php:51
/home/weirdan/src/psalm/psalm/src/Psalm/Internal/Analyzer/Statements/Expression/Call/ArgumentsAnalyzer.php:200
/home/weirdan/src/psalm/psalm/src/Psalm/Internal/Analyzer/Statements/Expression/Call/FunctionCallAnalyzer.php:149
/home/weirdan/src/psalm/psalm/src/Psalm/Internal/Analyzer/Statements/ExpressionAnalyzer.php:266
/home/weirdan/src/psalm/psalm/src/Psalm/Internal/Analyzer/Statements/ExpressionAnalyzer.php:51
/home/weirdan/src/psalm/psalm/src/Psalm/Internal/Analyzer/Statements/Expression/AssignmentAnalyzer.php:216
/home/weirdan/src/psalm/psalm/src/Psalm/Internal/Analyzer/Statements/ExpressionAnalyzer.php:135
/home/weirdan/src/psalm/psalm/src/Psalm/Internal/Analyzer/Statements/ExpressionAnalyzer.php:51
/home/weirdan/src/psalm/psalm/src/Psalm/Internal/Analyzer/StatementsAnalyzer.php:552
/home/weirdan/src/psalm/psalm/src/Psalm/Internal/Analyzer/StatementsAnalyzer.php:186
/home/weirdan/src/psalm/psalm/src/Psalm/Internal/Analyzer/FileAnalyzer.php:190
/home/weirdan/src/psalm/psalm/tests/TestCase.php:134
/home/weirdan/src/psalm/psalm/tests/Traits/ValidCodeAnalysisTestTrait.php:83
```

Reply:
So the ``PHP72-`` prefix is correct then.